### ISSUE TITLE
Build arg regression with 0.44.0

### ISSUE THREAD

**Issue by mattnelson** [fabric8io/docker-maven-plugin#1756]
### Description

Build arg regression with 0.44.0, possibly from #1731

```
Caused by: com.google.gson.JsonSyntaxException: Expected a com.google.gson.JsonObject but was com.google.gson.JsonArray
    at com.google.gson.internal.bind.TypeAdapters$33$1.read (TypeAdapters.java:869)
    at com.google.gson.Gson.fromJson (Gson.java:963)
    at com.google.gson.Gson.fromJson (Gson.java:928)
    at com.google.gson.Gson.fromJson (Gson.java:877)
    at com.google.gson.Gson.fromJson (Gson.java:848)
    at io.fabric8.maven.docker.util.JsonFactory.newJsonObject (JsonFactory.java:18)
    at io.fabric8.maven.docker.access.hc.DockerAccessWithHcClient.getImageId (DockerAccessWithHcClient.java:478)
    at io.fabric8.maven.docker.service.QueryService.getImageId (QueryService.java:135)
    at io.fabric8.maven.docker.service.BuildService.autoPullBaseImage (BuildService.java:385)
    at io.fabric8.maven.docker.service.BuildService.buildImage (BuildService.java:69)
    at io.fabric8.maven.docker.BuildMojo.proceedWithDockerBuild (BuildMojo.java:106)
    at io.fabric8.maven.docker.BuildMojo.proceedWithBuildProcess (BuildMojo.java:89)
    at io.fabric8.maven.docker.BuildMojo.buildAndTag (BuildMojo.java:82)
    at io.fabric8.maven.docker.BuildMojo.processImageConfig (BuildMojo.java:159)
    at io.fabric8.maven.docker.BuildMojo.executeInternal (BuildMojo.java:71)
    at io.fabric8.maven.docker.AbstractDockerMojo.execute (AbstractDockerMojo.java:295)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
```

### Info

* docker-maven-plugin version : 0.44.0
* Maven version (`mvn -v`) : 3.9.6

* Docker version : Docker version 24.0.7-rd, build 72ffacf
* If it's a bug, how to reproduce :

Have a dockerfile that uses a build arg for the FROM image.
```dockerfile
ARG FROM_IMAGE
FROM ${FROM_IMAGE} AS jlink
```

Flex that build arg via the plugin configuration stanzas.
```xml
<plugin>
  <groupId>io.fabric8</groupId>
  <artifactId>docker-maven-plugin</artifactId>
  <configuration>
    <images>
      <image>
        <name>myimage:21</name>
        <build>
          <args>
            <FROM_IMAGE>openjdk:21</FROM_IMAGE>
          </args>
        </build>
      </image>
      <image>
        <name>myimage:17</name>
        <build>
          <args>
            <FROM_IMAGE>openjdk:17</FROM_IMAGE>
          </args>
        </build>
      </image>
    </images>
  </configuration>
</plugin>
```

### Partial Workaround

Running the build with `mvn package -DFROM_IMAGE=openjdk:21` works, but is only a viable workaround when building a single image for the entire build.

**Comment by rohanKanojia** [comment]
@mattnelson : Thanks for reporting, I can reproduce this issue. I'll try to fix it .

**Comment by rohanKanojia** [comment]
In #1731 source of buildArgs was changed to use build args from maven/system properties. But using that dropped the source from build configuration, which caused this regession.

**Comment by rohanKanojia** [comment]
@mattnelson : I've merged changes that are supposed to fix this issue.

Could you please try out [Docker Maven Plugin Snapshots](https://github.com/fabric8io/docker-maven-plugin/blob/master/doc/HowToUseDockerMavenPluginSnapshotArtifacts.md) and see if it fixes your issue.

**Comment by poikilotherm** [comment]
Fixes it for me.

Before:
```
[INFO] DOCKER> [gdcc/configbaker:unstable] "dev_bootstrap": Created docker-build.tar in 25 milliseconds
[ERROR] DOCKER> Unable to inspect image [solr:] : {"message":"invalid reference format"} (Bad Request: 400) [{"message":"invalid reference format"} (Bad Request: 400)]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  15.175 s
[INFO] Finished at: 2024-04-12T13:36:10+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.fabric8:docker-maven-plugin:0.44.0:build (default-build) on project dataverse: Unable to inspect image [solr:] : {"message":"invalid reference format"} (Bad Request: 400) -> [Help 1]
```

With `0.45-SNAPSHOT`:
```
[INFO] DOCKER> [gdcc/configbaker:unstable] "dev_bootstrap": Built image sha256:6c72fdb9d53b0065597c409e54c599c11a91c56213c28e04136d9d7ef8eb9bd2
[INFO] DOCKER> gdcc/configbaker:unstable: Removed dangling image sha256:f9b1305613f99d577d81ed3e93739cdbb16a324b5b60d7800c0752d758d8f447
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  15.945 s
[INFO] Finished at: 2024-04-12T13:36:39+02:00
[INFO] ------------------------------------------------------------------------
```

**Comment by rohanKanojia** [comment]
@poikilotherm : Thanks for trying it out. I also need to check https://github.com/fabric8io/docker-maven-plugin/issues/1778 if it's related, then I'll try to cut a new release.

**Comment by ggjulio** [comment]
Same regression from `0.43.0` to `0.44.0`.

Can the issue be reopened ?

```log
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.fabric8:docker-maven-plugin:0.44.0:build (default) on project [PROJECT_NAME]: Execution default of goal io.fabric8:docker-maven-plugin:0.44.0:build failed: Expected a com.google.gson.JsonObject but was com.google.gson.JsonArray
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
	at org.codehaus.classworlds.Launcher.main(Launcher.java:47)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default of goal io.fabric8:docker-maven-plugin:0.44.0:build failed: Expected a com.google.gson.JsonObject but was com.google.gson.JsonArray
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:145)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	... 21 more
Caused by: com.google.gson.JsonSyntaxException: Expected a com.google.gson.JsonObject but was com.google.gson.JsonArray
	at com.google.gson.internal.bind.TypeAdapters$33$1.read(TypeAdapters.java:869)
	at com.google.gson.Gson.fromJson(Gson.java:963)
	at com.google.gson.Gson.fromJson(Gson.java:928)
	at com.google.gson.Gson.fromJson(Gson.java:877)
	at com.google.gson.Gson.fromJson(Gson.java:848)
	at io.fabric8.maven.docker.util.JsonFactory.newJsonObject(JsonFactory.java:18)
	at io.fabric8.maven.docker.access.hc.DockerAccessWithHcClient.getImageId(DockerAccessWithHcClient.java:478)
	at io.fabric8.maven.docker.service.QueryService.getImageId(QueryService.java:135)
	at io.fabric8.maven.docker.service.BuildService.autoPullBaseImage(BuildService.java:385)
	at io.fabric8.maven.docker.service.BuildService.buildImage(BuildService.java:69)
	at io.fabric8.maven.docker.BuildMojo.proceedWithDockerBuild(BuildMojo.java:106)
	at io.fabric8.maven.docker.BuildMojo.proceedWithBuildProcess(BuildMojo.java:89)
	at io.fabric8.maven.docker.BuildMojo.buildAndTag(BuildMojo.java:82)
	at io.fabric8.maven.docker.BuildMojo.processImageConfig(BuildMojo.java:159)
	at io.fabric8.maven.docker.BuildMojo.executeInternal(BuildMojo.java:71)
	at io.fabric8.maven.docker.AbstractDockerMojo.execute(AbstractDockerMojo.java:295)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	... 22 more

```

**Comment by rohanKanojia** [comment]
@ggjulio : I haven't released a new version since this change got merged.

You might want to verify on 0.45-SNAPSHOT ( try [Docker Maven Plugin Snapshots](https://github.com/fabric8io/docker-maven-plugin/blob/master/doc/HowToUseDockerMavenPluginSnapshotArtifacts.md) and see if it fixes your issue.)

**Comment by ggjulio** [comment]
Nice, it fixes the issue. Thank you !

**Comment by rohanKanojia** [comment]
@ggjulio : Sorry for the delay, I'll try to cut 0.45.0 by the end of this month.

**Comment by ggjulio** [comment]
Hi @rohanKanojia, do you have a new estimated date for the next release ?
Thanks !

**Comment by rohanKanojia** [comment]
@ggjulio : I'm really sorry for the delay. I will try to cut a release this week.

**Comment by rohanKanojia** [comment]
@ggjulio : : I have released [0.45.0](https://repo1.maven.org/maven2/io/fabric8/docker-maven-plugin/0.45.0/) . Could you please check if everything is working as expected in your project and provide feedback?
